a8c-files.php

```diff
@@ -776,6 +776,232 @@ function a8c_files_init() {
 	new A8C_Files();
 }
 
+class A8C_Files_Image_Srcset_Meta {
+
+	/**
+	 * Create proper 'sizes' array for the image meta including the Photon query args used on the img src.
+	 *
+	 * @param array  $image_meta    The image meta data as returned by 'wp_get_attachment_metadata()'.
+	 * @param array  $size_array    Array of width and height values in pixels (in that order).
+	 * @param string $image_src     The 'src' of the image.
+	 * @param int    $attachment_id The image attachment ID or 0 if not supplied.
+	 * @return array                The normalized image meta data or empty $image_meta['sizes'] on error.
+	 */
+	public static function generate_image_meta( $image_meta, $size_array, $image_src, $attachment_id ) {
+		// Reset $image_meta['sizes'] if not empty. Need to build it again matching the current img src.
+		$image_meta['sizes'] = array();
+
+		// Bail out early if the image is not hosted on the site, missing image meta or original image is too small.
+		// This will disable the generation of srcset.
+		if ( strpos( $image_src, home_url() ) === false ||
+			empty( $image_meta['width'] ) ||
+			empty( $image_meta['height'] ) ||
+			( $image_meta['width'] < 100 && $image_meta['height'] < 100 ) ) {
+
+			return $image_meta;
+		}
+
+		$image_basename = wp_basename( $image_meta['file'] );
+
+		// Return early if filename mismatch
+		if ( strpos( wp_basename( $image_src ), $image_basename ) === false ) {
```

Comment:
To make it easier to follow what's being compared with what, the second `wp_basename()` call should also be set in a variable. I'm also curious why we don't just compare the two results. Why is the `strpos()` necessary?